src/components/ToDoItemComp.vue

```diff
@@ -1,32 +1,223 @@
+                  d="M19 9l-7 7-7-7"
+                ></path>
+              </svg>
+            </button>
+          </div>
+          <!-- Dropdown menu -->
+          <div
+            v-if="showDropDown"
+            id="dropdown"
+            class="cursor-pointer absolute z-10 top-12 text-lg font-semibold rounded-xl border-black border-2 text-start items-center"
+          >
+            <ul class="py-2 pl-4 pr-3" aria-labelledby="dropdownButton">
+              <li v-for="priority in priorityList">
+                <p class="opacity-100" @click="changePriority(priority)">
+                  {{ priority }}
+                </p>
+              </li>
+            </ul>
+          </div>
+        </div>
+      </div>
+      <div
+        class="flex flex-row space-x-1"
+        :class="showDropDown ? 'opacity-30' : ''"
+      >
+        <img src="../assets/icons/Vectorcalendar.svg" />
+        <p class="text-xs">{{ formatShortDate(date) }}</p>
+      </div>
+      <div class="self-start mt-4 mb-4">
+        <textarea
+          v-model="currentText"
+          placeholder="Lorem ipsum"
+          class="border-none h-[70.27px] w-[224.45px] lg:h-[103px] lg:w-[455.1px] text-start text-sm lg:text-lg text-grey"
+          :class="showDropDown ? 'opacity-30' : ''"
+          :disabled="showDropDown"
+        ></textarea>
+      </div>
+      <div class="text-sm flex flex-row font-semibold space-x-2">
+        <button
+          class="text-white w-[61px] h-[27px] rounded-lg bg-green"
+          :class="showDropDown ? 'opacity-30' : ''"
+          :disabled="showDropDown"
+          @click="saveToDo"
+        >
+          Save
+        </button>
+        <button
+          class="rounded-lg h-[27px] w-[70px] bg-buttonGray"
+          :class="showDropDown ? 'opacity-30' : ''"
+          :disabled="showDropDown"
+          @click="deleteToDo"
+        >
+          Delete
+        </button>
+      </div>
     </div>
   </div>
 </template>
 
 <script setup lang="ts">
-  import { ToDoItem } from '../models/todoitem-model';
+  import { computed, ref } from 'vue';
+  import { Priority, ToDoItem, Status } from '../models/todoitem-model';
+  import { formatShortDate } from '../utils/date-formatting';
+  import { getItemPriority } from '../utils/item-priority';
 
   const props = defineProps<{
-    item: ToDoItem;
-    index: number;
+    item?: ToDoItem;
+    index?: number;
   }>();
 
   const emit = defineEmits<{
     (e: 'removeItem', ind: number): void;
+    (e: 'addToDo', newToDo: ToDoItem): void;
+    (e: 'hideForm'): void;
+    (e: 'updateToDo', ind: number, newToDo: ToDoItem): void;
   }>();
 
-  function removeTodo(ind: number) {
-    emit('removeItem', ind);
+  const showDropDown = ref(false);
+
+  const currentTitle = ref(props.item ? props.item.title : '');
+
+  const currentText = ref(props.item ? props.item.text : '');
+
+  const indexOfThisItem = ref(
+    props.index || props.index === 0 ? props.index : -1
```

Comment:
You can optimise this if you add a default value for index prop to be -1 
See [Vue props](https://vuejs.org/guide/components/props.html#prop-validation)